### PR TITLE
DIA-4610 Add KMM's `customConsent` and `deleteCustomConsent`

### PR DESCRIPTION
**Pull request by Nevazhnovu**
[DIA-4610](https://sourcepoint.atlassian.net/browse/DIA-4610) [iOS] Integrate custom-consent call

[DIA-4610]: https://sourcepoint.atlassian.net/browse/DIA-4610?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Example/ConsentViewController_ExampleTests/SPClientCoordinator/SourcePointClient/SourcePointClientSpec.swift [hunk 1]
```diff
@@ -329,80 +329,6 @@ class SourcePointClientSpec: QuickSpec {
             }
 
             describe("customConsent") {
```
Comment (andresilveirah) [inline Example/ConsentViewController_ExampleTests/SPClientCoordinator/SourcePointClient/SourcePointClientSpec.swift:331]: @Nevazhnovu I believe we can delete the entire `describe` block, including that test about the `custom-metrics`. That's already covered in the core-module.

File: ConsentViewController/Classes/SourcePointClient/SourcePointClient.swift [hunk 2]
```diff
@@ -367,7 +422,7 @@ class SourcePointClient: SourcePointProtocol {
         }
     }
 
-    func deleteCustomConsentGDPR(
+    func fallbackDeleteCustomConsentGDPR(
```
Comment (andresilveirah) [inline ConsentViewController/Classes/SourcePointClient/SourcePointClient.swift:425]: I think we're safe without a fallback. Let's remove it.

File: ConsentViewController/Classes/SourcePointClient/SourcePointClient.swift [hunk 3]
```diff
@@ -348,6 +348,61 @@ class SourcePointClient: SourcePointProtocol {
...
+                        self.fallbackCustomConsentGDPR(
+                            toConsentUUID: consentUUID,
+                            vendors: vendors,
+                            categories: categories,
+                            legIntCategories: legIntCategories,
+                            propertyId: propertyId,
+                            handler: handler
+                        )
+                    } else {
+                        handler(Result.success(response!.toNativeAsAddOrDeleteCustomConsentResponse())) // swiftlint:disable:this force_unwrapping
```
Comment (andresilveirah) [inline ConsentViewController/Classes/SourcePointClient/SourcePointClient.swift:373]: Is it safe to unwrap here?

Comment (Nevazhnovu) [reply ConsentViewController/Classes/SourcePointClient/SourcePointClient.swift:373]: It should be. We check the response for being null in the `if` statement above

**Review by andresilveirah — APPROVED** [review]

**Pull request merged**